### PR TITLE
separate user page and welcome page (from slack links)

### PR DESCRIPTION
**Pull request by camhashemi**
@ryanpbrewster subtle differences between welcome and user page...

### CODE REVIEW

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala [hunk 1]
```diff
@@ -50,17 +50,17 @@ class SlackAuthRouter @Inject() (
...
-          userPath <- userPathOpt
-          orgId <- slackTeamRepo.getBySlackTeamId(slackTeamId).flatMap(_.organizationId)
-          org <- orgRepo.getActive(orgId).orElse { airbrake.notify(s"[zombie-org] slackTeam=${slackTeamId.value} references inactive org=${orgId}"); None }
-        } yield redirectThroughSlackAuth(org, slackTeamId, userPath, userId = Some(extId))
-      }
+      (for {
+        _ <- Some(true) if request.userIdOpt.isEmpty || isWelcomeMessage
+        userPath <- userPathOpt
+        orgId <- slackTeamRepo.getBySlackTeamId(slackTeamId).flatMap(_.organizationId)
+        org <- orgRepo.getActive(orgId).orElse { airbrake.notify(s"[inactive-org] slackTeam=${slackTeamId.value} references inactive org=${orgId.id}"); None }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala]: Neat!

I wonder if `tap` is cleaner here.
`org <- orgRepo.getActive(orgId) tap { v => if(v.isEmpty) airbrake.notify(s"$slackTeamId has inactive org $orgId") }`
Unclear to me that it's actually better.

I almost want a `tapWhere` that takes a partial function.

**Comment by ryanpbrewster** [comment]
:+1: lgtm

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala (hunk 1 above)
Comment (camhashemi) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala]: yeah, that is better :) partial function inherits from function so we good

File: kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala [hunk 2]
```diff
@@ -50,17 +50,21 @@ class SlackAuthRouter @Inject() (
...
-        } yield redirectThroughSlackAuth(org, slackTeamId, userPath, userId = Some(extId))
-      }
+      (for {
+        _ <- Some(true) if request.userIdOpt.isEmpty || isWelcomeMessage
+        userPath <- userPathOpt
+        orgId <- slackTeamRepo.getBySlackTeamId(slackTeamId).flatMap(_.organizationId)
+        org <- orgRepo.getActive(orgId).tap {
+          case None => airbrake.notify(s"[inactive-org] slackTeam=${slackTeamId.value} references inactive org=${orgId.id}")
+          case _ =>
+        }
```
Comment (ryanpbrewster) [inline kifi-backend/modules/shoebox/app/com/keepit/controllers/routing/SlackAuthRouter.scala:65]: :boom: nice

**Pull request merged**